### PR TITLE
Fixes #399 String.characters.count deprecation warning in Xcode 9.1

### LINKED ISSUE #399: String characters count deprecation
When I integrate Starscream in my lib and run `pod lib lint` I get deprecation warnings:

```
- WARN  | xcodebuild:  Starscream/Sources/WebSocket.swift:534:35: warning: 'characters' is deprecated: Please use String or Substring directly
- WARN  | xcodebuild:  Starscream/Sources/WebSocket.swift:810:26: warning: 'characters' is deprecated: Please use String or Substring directly
- WARN  | xcodebuild:  Starscream/Sources/WebSocket.swift:811:33: warning: 'characters' is deprecated: Please use String or Substring directly
```

### PR DESCRIPTION
**Pull request by fassko**
String.characters.count deprecation warnings

### CODE REVIEW

File: Sources/WebSocket.swift [hunk 1]
```diff
@@ -531,7 +531,7 @@ open class WebSocket : NSObject, StreamDelegate, WebSocketClient, WSStreamDelega
         request.setValue("\(url.host!):\(port!)", forHTTPHeaderField: headerWSHostName)
 
         var path = url.absoluteString
-        let offset = (url.scheme?.characters.count ?? 2) + 3
+        let offset = (url.scheme?.count ?? 2) + 3
```
Comment (dfed) [inline Sources/WebSocket.swift]: This change will break for anyone not yet on Xcode 9. Might be best to do the following instead:
```
#if swift(>=3.2)
let offset = (url.scheme?.count ?? 2) + 3
#else
let offset = (url.scheme?.characters.count ?? 2) + 3
#endif
```

Comment (dfed) [reply Sources/WebSocket.swift]: Nevermind just saw we're already Swift 4 🙂

Comment (fassko) [reply Sources/WebSocket.swift]: This deprecation message actually is only in Xcode 9.1 but let's get ready for it. ;)

**Comment by daltoniam** [comment]
Thanks! Let us get ahead of the curve for once 😄.

**Pull request merged**